src/switch_rtp.c

```diff
@@ -3420,29 +3420,89 @@ SWITCH_DECLARE(switch_status_t) switch_rtp_set_remote_address(switch_rtp_t *rtp_
 	return status;
 }
 
-SWITCH_DECLARE(switch_status_t) switch_rtp_fork_set(switch_rtp_t *rtp_session, switch_fork_direction_t direction, const char *host, switch_port_t port, uint32_t ssrc, const char *cmd)
+SWITCH_DECLARE(switch_status_t) switch_rtp_fork_set(switch_rtp_t *rtp_session, switch_fork_direction_t direction, const char *host, switch_port_t port, uint32_t ssrc, const char *cmd, const char *codec_iananame)
 {
 	switch_sockaddr_t *addr = NULL;
 	switch_fork_session_t *fork = NULL;
+	int transcoding = 0;
+	switch_channel_t *channel = NULL;
+	char rtp_codec[FORK_CODEC_NAME_LEN] = { 0 };
+	char fork_codec[FORK_CODEC_NAME_LEN] = { 0 };
 
 	if (!rtp_session) {
 		switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(rtp_session->session), SWITCH_LOG_ERROR, "Fork: no RTP session\n");
 		return SWITCH_STATUS_FALSE;
 	}
 
+	channel = switch_core_session_get_channel(rtp_session->session);
+
 	fork = (direction == FORK_DIRECTION_RX ? &rtp_session->fork.fork_rx : &rtp_session->fork.fork_tx);
 
 	if (switch_sockaddr_info_get(&addr, host, SWITCH_UNSPEC, port, 0, rtp_session->pool) != SWITCH_STATUS_SUCCESS || !addr) {
-		switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(rtp_session->session), SWITCH_LOG_ERROR, "Fork: cannot reslove IP address\n");
+		switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(rtp_session->session), SWITCH_LOG_ERROR, "Fork: cannot resolve IP address\n");
 		return SWITCH_STATUS_FALSE;
 	}
 
+
+	// Transcode audio to requested fork codec if necessary
+	if (!zstr(codec_iananame)) {
+
+		const char *rtp_codec_iananame = NULL;
+		uint32_t rate_in = 8000, rate_out = 8000;
+		uint32_t bitrate_in = 0, bitrate_out = 0;
+		uint32_t channels_in = 1, channels_out = 1;
+		uint32_t ptime_in = 20, ptime_out = 20;
+
+		strncpy(fork_codec, codec_iananame, 40);
```

Comment:
Should we lower the length of fork_codec or rtp_codec since we only copy just 40 bytes?

Reply:
It is different in amended commit anyway.